Sources/SwiftProtobuf/MathUtils.swift

```diff
@@ -38,3 +38,98 @@ internal func div<T : SignedInteger>(_ a: T, _ b: T) -> T {
     assert(b > 0)
     return a >= 0 ? a / b : (a + 1) / b - 1
 }
+
+// Swift's standard library changed the name of the initializer
+// that truncates an integer to a narrower type.  To continue
+// to support Swift 3, we've introduced these functions to
+// handle that.
+//
+#if swift(>=4.0)
```

Comment:
So rather then having something we have to go touch the code to clean up, why not follow what we did Int64 and UInt64. There we provided the init?(exactly:) for older versions of swift, and nothing was needed going forward.  This means no shim on newer versions, and the "extra" code compiles away until we trim it.  - https://github.com/apple/swift-protobuf/blob/master/Sources/SwiftProtobuf/JSONIntegerConverting.swift#L18